code/modules/vending/_vending.dm

```diff
@@ -532,7 +532,7 @@ GLOBAL_LIST_EMPTY(vending_products)
 
 ///Tilts ontop of the atom supplied, if crit is true some extra shit can happen. Returns TRUE if it dealt damage to something.
 /obj/machinery/vending/proc/tilt(atom/fatty, crit=FALSE)
-	if(QDELETED(src))
+	if(QDELETED(src) || !has_gravity(src))
```

Comment:
you should probably put this in the melee vending machine fighting, as tilting also happens when a vending machine is thrown at someone

Reply:
> you should probably put this in the melee vending machine fighting, as tilting also happens when a vending machine is thrown at someone

do you mean like when someone's baseball batted towards it? i tested that its handled here